### ISSUE TITLE
Update crashes after migration from 3.3.0 to 3.4.2

### ISSUE THREAD

**Issue by renaudk** [osm-search/Nominatim#1855]
Hello,

The migration guide indicates the following commands for upgrade :
```
CREATE TABLE location_area_country_new AS
  SELECT place_id, country_code, geometry FROM location_area_country;
DROP TABLE location_area_country;
ALTER TABLE location_area_country_new RENAME TO location_area_country;
CREATE INDEX idx_location_area_country_geometry ON location_area_country USING GIST (geometry);
CREATE INDEX idx_location_area_country_place_id ON location_area_country USING BTREE (place_id);
```

If I run theses commands, the update fails with error :
```
Starting rank 2
index_placex: UPDATE failed: ERROR:  column "rank_search" does not exist
LINE 4: ...ountry WHERE ST_Intersects(geometry, feature) and rank_searc...
                                                             ^
HINT:  There is a column named "rank_search" in table "*SELECT* 1", but it cannot be referenced from this part of the query.
QUERY:  SELECT place_id, keywords, rank_address, rank_search, min(ST_Distance(feature, centroid)) as distance, isguess, postcode, centroid FROM (
        SELECT * FROM location_area_large_0 WHERE ST_Intersects(geometry, feature) and rank_search < maxrank
        UNION ALL
        SELECT * FROM location_area_country WHERE ST_Intersects(geometry, feature) and rank_search < maxrank
      ) as location_area
      GROUP BY place_id, keywords, rank_address, rank_search, isguess, postcode, centroid
      ORDER BY rank_address, isin_tokens && keywords desc, isguess asc,
        ST_Distance(feature, centroid) *
          CASE
               WHEN rank_address = 16 AND rank_search = 15 THEN 0.2 -- capital city
               WHEN rank_address = 16 AND rank_search = 16 THEN 0.25 -- city
               WHEN rank_address = 16 AND rank_search = 17 THEN 0.5 -- town
               ELSE 1 END ASC -- everything else
CONTEXT:  PL/pgSQL function getnearfeatures(integer,geometry,integer,integer[]) line 5258 at FOR over SELECT rows
PL/pgSQL function placex_update() line 613 at FOR over SELECT rows
Error: 1
```

I'm using ZFS so, I just rolled back databse to point before altering with migration commands above, and running update is OK.

Is there a mistake in migration commands ?

BR,

**Comment by mtmail** [comment]
You can inspect the function source code of `getNearFeatures` in the database by running
```
\df+ getNearFeatures
```

Check if these changes from 9 Jul 2019 are applied: https://github.com/osm-search/Nominatim/commit/be47cd2549da482f3a408307eee8bfe7967cff25

If not:
```
./utils/setup.php --create-functions
```

This will execute
https://github.com/osm-search/Nominatim/blob/be47cd2549da482f3a408307eee8bfe7967cff25/sql/functions.sql
against the database (first it replaces some placeholders with values from the configuration file). The `CREATE OR REPLACE FUNCTION ...` statements update the source code.

**Comment by renaudk** [comment]
The function getNearFeatures is the old one.

Event if I run `./utils/setup.php --create-functions` getNearFeatures is not updated.

Maybe I missed something in the upgrade process.
I just ran the build: cmake, make, make install and updated all paths and local.php file according to my installation. Then I ran the migration SQL commands.
Finally i ran `utils/update.php --init-updates` then restarted updates.
